### PR TITLE
Add autoIncremented parameter to batch-post in bmqtool

### PR DESCRIPTION
**Pull request by syuzvinsky**
A new parameter 'autoIncremented' can be provided to 'batch-post' command in bmqtool.
It can be used to test subscriptions with expressions like 'x % 3 == 0' where "x" is a value of 'autoIncremented' field, so, every 3rd element will be received by the subscriber.

### CODE REVIEW

File: src/applications/bmqtool/m_bmqtool_parameters.h [hunk 1]
```diff
@@ -358,9 +363,8 @@ class Parameters {
     int                                 shutdownGrace() const;
     bool                                noSessionEventHandler() const;
     const bsl::vector<MessageProperty>& messageProperties() const;
-
-    /// Return the corresponding data member value.
-    const bsl::vector<Subscription>& subscriptions() const;
+    const bsl::vector<Subscription>&    subscriptions() const;
+    bsl::string                         autoIncrementedField() const;
```
Comment (678098) [inline src/applications/bmqtool/m_bmqtool_parameters.h:367]:
```suggestion
    const bsl::string&                  autoIncrementedField() const;
```

File: src/applications/bmqtool/m_bmqtool_poster.h [hunk 2]
```diff
@@ -95,6 +95,10 @@ class PostingContext {
     bmqa::MessageProperties d_properties;
     // Properties that will be added to a posted message
 
+    unsigned int d_autoIncrementedValue;
```
Comment (678098) [inline src/applications/bmqtool/m_bmqtool_poster.h:98]:
```suggestion
    bsls::Types::Uint64 d_autoIncrementedValue;
```

File: src/applications/bmqtool/m_bmqtool_poster.h [hunk 3]
```diff
@@ -95,6 +95,10 @@ class PostingContext {
...
+    // A value that will be auto-incremented and added to
+    // the message properties.
```
Comment (678098) [inline src/applications/bmqtool/m_bmqtool_poster.h:100]: Might convert to Doxygen style comment

**Review by 678098 — CHANGES_REQUESTED** [review]

File: src/applications/bmqtool/m_bmqtool_parameters.h [hunk 4]
```diff
@@ -268,6 +268,10 @@ class Parameters {
 
     bsl::vector<Subscription> d_subscriptions;
 
+    bsl::string d_autoIncrementedField;
+    // A name of a property to put auto-incremented values
+    // in batch-posting mode.
```
[the remark this answers is not in the document]
Comment (syuzvinsky) [reply src/applications/bmqtool/m_bmqtool_parameters.h:273]: The style is derived from the style used for all private variables in this file. Applying doxygen style for just one field looks impractical.

File: src/applications/bmqtool/m_bmqtool_poster.h (hunk 3 above)
Comment (syuzvinsky) [reply src/applications/bmqtool/m_bmqtool_poster.h:100]: the same comment about style uniformity

File: src/applications/bmqtool/m_bmqtool_parameters.h [hunk 5]
```diff
@@ -705,6 +716,11 @@ inline const bsl::vector<Subscription>& Parameters::subscriptions() const
     return d_subscriptions;
 }
 
+inline bsl::string Parameters::autoIncrementedField() const
```
Comment (678098) [inline src/applications/bmqtool/m_bmqtool_parameters.h:719]:
```suggestion
inline
const bsl::string &Parameters::autoIncrementedField() const
```

File: src/applications/bmqtool/m_bmqtool_parameters.h [hunk 6]
```diff
@@ -705,6 +716,12 @@ inline const bsl::vector<Subscription>& Parameters::subscriptions() const
     return d_subscriptions;
 }
 
+inline
+const bsl::string &Parameters::autoIncrementedField() const
```
Comment (678098) [inline src/applications/bmqtool/m_bmqtool_parameters.h:720]: Making formatter happy
```suggestion
inline const bsl::string &Parameters::autoIncrementedField() const
```

**Review by 678098 — APPROVED** [review]
LGTM!

**Pull request merged**